src/views/RegionSeasonalRiskWatch/Filters/index.tsx

```diff
@@ -54,13 +56,48 @@ function Filters(props: Props) {
 
     const countryList = useCountry({ region: regionId });
 
+    useEffect(
+        () => {
+            // TODO: figure out better way to set default value
+            // NOTE: setting default value
+            onChange((prevValue) => ({
+                ...prevValue,
+                countries: countryList.map((country) => country.iso3),
+                months: Array.from(Array(11).keys()),
+                hazardTypes: mapToList(
+                    applicableHazardsByRiskMetric[prevValue.riskMetric],
+                    (isApplicable, hazardType) => (isApplicable
+                        ? hazardType as HazardType : undefined),
+                ).filter(isDefined),
+            }));
+        },
+        [countryList, onChange],
+    );
+
     const handleChange = useCallback(
         (...args: EntriesAsList<FilterValue>) => {
             const [val, key] = args;
-            onChange((prevValue): FilterValue => ({
-                ...prevValue,
-                [key]: val,
-            }));
+            onChange((prevValue): FilterValue => {
+                if (key !== 'riskMetric') {
+                    return {
+                        ...prevValue,
+                        [key]: val,
+                    };
+                }
+
+                const applicableHazards = applicableHazardsByRiskMetric[val as RiskMetric];
+                const hazardTypes = mapToList(
+                    applicableHazards,
+                    (isApplicable, hazardType) => (isApplicable
+                        ? hazardType as HazardType : undefined),
+                ).filter(isDefined);
+
+                return {
+                    ...prevValue,
+                    riskMetric: val as RiskMetric,
+                    hazardTypes,
+                };
```

Comment:
Let's create a separate handleChange for 'riskMetric'
We wouldn't need to cast `val` as RiskMetric that way